pkg/library/home/persistentHome.go

```diff
@@ -95,3 +99,77 @@ func storageStrategySupportsPersistentHome(workspace *common.DevWorkspaceWithCon
 	storageClass := workspace.Spec.Template.Attributes.GetString(constants.DevWorkspaceStorageTypeAttribute, nil)
 	return storageClass != constants.EphemeralStorageClassType
 }
+
+func addInitContainer(dwTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec, initContainerFromConfig *v1alpha2.Container) {
+
+	initContainerComponent := v1alpha2.Component{
+		Name: constants.HomeInitComponentName,
+	}
+
+	initContainer := initContainerFromConfig
+
+	if initContainer == nil {
+		initContainer = inferInitContainer(dwTemplateSpec)
+	}
+
+	if initContainer == nil {
+		return
+	}
+
+	initContainerComponent.ComponentUnion = v1alpha2.ComponentUnion{
+		Container: &v1alpha2.ContainerComponent{
+			Container: *initContainer,
+		},
+	}
+
+	dwTemplateSpec.Components = append(dwTemplateSpec.Components, initContainerComponent)
+	dwTemplateSpec.Commands = append(dwTemplateSpec.Commands, v1alpha2.Command{
+		Id: constants.HomeInitEventId,
+		CommandUnion: v1alpha2.CommandUnion{
+			Apply: &v1alpha2.ApplyCommand{
+				Component: constants.HomeInitComponentName,
+			},
+		},
+	})
+	dwTemplateSpec.Events.PreStart = append(dwTemplateSpec.Events.PreStart, constants.HomeInitEventId)
+}
+
+func inferInitContainer(dwTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec) *v1alpha2.Container {
+	var nonImportedComponent v1alpha2.Component
+	for _, component := range dwTemplateSpec.Components {
+		if component.Container == nil {
+			continue
+		}
+
+		pluginSource := component.Attributes.GetString(constants.PluginSourceAttribute, nil)
+		if pluginSource == "" || pluginSource == "parent" {
+			// First, non-imported container component is selected
+			nonImportedComponent = component
+			break
+		}
+	}
+
+	createOutFile := "touch /home/user/.init-persistent-home"
```

Comment:
Is there an underlying reason for putting the `.init-persistent-home` script in the /home/user/ directory? Maybe it'd be better to keep it in the root directory `/` (similar to the entrypoint, though we might not have write access there...) or in` /tmp/` so that users don't notice a shell script being added to their $HOME directory?

Reply:
My mistake, I misinterpreted what we were doing here. I now see that `/home/user/.init-persistent-home` is supposed to be the command output of the user-provided `$HOME_SETUP_SCRIPT` or the default `runStow` script. 

I still think it might make more sense to put the `.init-persistent-home` into /tmp/ (since we put the pre and post-start command outputs as well as the stow command output there). It also would make sense to not make it hidden, i.e. the file should be `/tmp/init-persistent-home` instead of `/tmp/.init-persistent-home/`.

Reply:
Sorry, this functionality was just a small idea, I've removed it in my latest commits